src/sql/connection.py

```diff
@@ -447,5 +460,103 @@ def get_curr_identifiers(cls) -> list:
                 identifiers = [*set(identifiers + identifiers_)]
         except ValueError:
             pass
+        except AttributeError:
+            # this might be a custom connection..
+            pass
 
         return identifiers
+
+    @classmethod
+    def is_custom_connection(self, conn=None) -> bool:
+        """
+        Checks if given connection is custom
+        """
+        is_custom_connection_ = False
+
+        if conn is None:
+            if not Connection.current:
+                raise RuntimeError("No active connection")
+            else:
+                conn = Connection.current.session
+
+        if isinstance(conn, (CustomConnection, CustomSession)):
+            is_custom_connection_ = True
+        else:
+            # TODO: Better check when user passes a custom
+            # connection
+            is_custom_connection_ = conn.__class__.__name__ == "connection"
+        return is_custom_connection_
+
+    @classmethod
+    def prepare_query(self, query, conn=None, with_=None) -> str:
```

Comment:
I think this can be changed to non classmethod too, it's more related to runtime engine